plugins/search-backend-node/src/types.ts

```diff
@@ -57,3 +60,55 @@ export interface RegisterDecoratorParameters {
    */
   factory: DocumentDecoratorFactory;
 }
+
+/**
+ * A type of function responsible for translating an abstract search query into
+ * a concrete query relevant to a particular search engine.
+ * @public
+ */
+export type QueryTranslator = (query: SearchQuery) => unknown;
+
+/**
+ * Options when querying a search engine.
+ * @public
+ */
+export type QueryRequestOptions =
+  | {
+      /** @deprecated use the `credentials` option instead. */
+      token?: string;
```

Comment:
I am wondering if it is more dangerous to leave this one or remove it completely. Thinking that the build will pass for any plugin still using `token` (although there is none in `backstage` AFAIK), but fail at runtime as `credentials` are passed from the `AuthorizedSearchEngine`. Or rather, start doing unauthorized calls at runtime.

Not sure if this is a problem or if it's simply offset by the details in the changeset?

Reply:
I'm not sure if there are custom adopters' search engines using this token, so that is the reason I kept it here.
For the search engines we have here in the main monorepo, only the `AuthorizedSearchEngine` uses these `options`, so in theory, none of OSS search engines use credentials or token here.
I wonder what should be the conditional to pass a token or credentials to the engine query method 🤔 ? Any thoughts @Rugvip or @Rugvip ?

Reply:
Also this is for keeping the old backend system supported still for a while, right?

We're still finalizing how all this will look, but ideally we want credentials to be passed around and then the lower levels of code will be able to transform those to an actual outgoing request service token when they can tell the auth service what the target plugin ID is that it wants to talk to etc